kong/clustering/rpc/callbacks.lua

```diff
@@ -0,0 +1,45 @@
+local _M = {}
+local _MT = { __index = _M, }
+
+
+local utils = require("kong.clustering.rpc.utils")
+
+
+local parse_method_name = utils.parse_method_name
+
+
+function _M.new()
+  local self = {
+    callbacks = {},
+    capabilities = {}, -- updated as register() is called
+    capabilities_list = {}, -- updated as register() is called
+  }
+
+  return setmetatable(self, _MT)
+end
+
+
+function _M:register(method, func)
+  if self.callbacks[method] then
+    error("duplicate registration of " .. method)
+  end
+
+  local cap, func_or_err = parse_method_name(method)
+  if not cap then
+    return nil, "unable to get capabilities: " .. func_or_err
+  end
+
+  self.capabilities[cap] = true
+  table.insert(self.capabilities_list, cap)
```

Comment:
It may insert duplicated cap into to `self.capabilities_list` for the following calls, which results in `["kong.debug.log_level.v1", "kong.debug.log_level.v1"]`. Is it as expected?

```
kong/clustering//services/debug.lua:  manager.callbacks:register("kong.debug.log_level.v1.get_log_level", rpc_get_log_level)
kong/clustering//services/debug.lua:  manager.callbacks:register("kong.debug.log_level.v1.set_log_level", rpc_set_log_level)
```

Reply:
Yes, I think that we should add some checking logic, like:
```
if not self.capabilities[cap] then
  self.capabilities[cap] = true
  table.insert(self.capabilities_list, cap)
end
```